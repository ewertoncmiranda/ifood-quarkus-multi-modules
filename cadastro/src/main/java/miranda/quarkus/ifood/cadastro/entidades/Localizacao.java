package miranda.quarkus.ifood.cadastro.entidades;


import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Table(name = "localizacao")
public class Localizacao  extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id ;
    private Double latitude ;
    private Double longitude ;

}
