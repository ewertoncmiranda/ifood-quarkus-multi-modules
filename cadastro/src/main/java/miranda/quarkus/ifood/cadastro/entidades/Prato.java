package miranda.quarkus.ifood.cadastro.entidades;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
public class Prato  extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id ;
    private String nome ;
    private String descricao;
    private BigDecimal valor;

    @ManyToOne
    private Restaurante restaurante ;
}
