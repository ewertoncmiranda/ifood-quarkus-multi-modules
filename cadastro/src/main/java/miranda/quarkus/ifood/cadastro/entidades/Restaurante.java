package miranda.quarkus.ifood.cadastro.entidades;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Restaurante extends PanacheEntityBase {
}
