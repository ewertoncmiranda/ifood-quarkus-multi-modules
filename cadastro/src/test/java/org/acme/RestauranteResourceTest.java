package org.acme;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import miranda.quarkus.ifood.cadastro.entidades.Restaurante;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

@DBRider
@QuarkusTest
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurante-cenario-1.yml")
    public void deveBuscarRestaurantes(){
        String result = RestAssured.given().when()
                .get("/restaurante")
                .then().statusCode(200).extract().asString();
        Approvals.verifyJson(result);
    }
    @Test
    @DataSet("restaurante-cenario-1.yml")
    public void deveAlterarRestaurante(){
        Restaurante dto = new Restaurante();
        dto.setNome("novoNome");;
        Long parameterValue = 123L;
        RestAssured.given().with().pathParam("id",parameterValue)
                .body(dto)
                .when().put("/restaurante/{id}")
                .then().statusCode(Response.Status.NO_CONTENT.getStatusCode()).extract().asString();

       Restaurante byId = Restaurante.findById(parameterValue);



        Assert.assertEquals(dto.getNome(),byId.getNome());
    }




}
