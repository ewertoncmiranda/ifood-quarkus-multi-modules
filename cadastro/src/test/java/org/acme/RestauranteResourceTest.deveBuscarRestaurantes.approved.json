[
  {
    "id": 1,
    "nome": "teste",
    "proprietario": "teste"
  },
  {
    "id": 2,
    "nome": "teste",
    "proprietario": "teste"
  }
]